### DISCUSSION TITLE
How to move a file and replace existing one?

### DISCUSSION THREAD

**Discussion by wiedymi** [qgustavor/mega#97] [Ideas]
How to move a file and replace existing one?

**Comment by qgustavor** [comment]
Moving files is documented here: https://github.com/qgustavor/mega/wiki/MutableFile#movetotarget-callback

This library does not support neither [version history](https://mega.nz/help/client/webclient/cloud-drive#can-i-manage-versions-of-my-files-in-mega-59f13b42f1b7093a7f8b4589) nor replacing existent files. You can create other file with the same name (which is something the web client doesn't allow) and delete the original file.

You can also submit a pull request implementing it. Seems the main difference is that [this object](https://github.com/qgustavor/mega/blob/a627db06be2d6d88ebac99906ee4340bc3a339d2/lib/mutable-file.js#L191-L195) needs to include some info about the original file. The code that handles that in the web client [is here](https://github.com/meganz/webclient/blob/2f2c3bde9e62a75ca2212da42d5faad66af32584/js/fm/fileversioning.js).

**Reply by danddinh** [reply]
how to delete the original file? i cant see it in the api reference at mega.js.org

**Reply by qgustavor** [reply]
https://mega.js.org/docs/1.0/api#deletepermanent ?
